v1.0/deploy-cockroachdb-to-production.md

```diff
@@ -0,0 +1,424 @@
+
+<div id="toc"></div>
+
+## Deployment Architecture
+
+To deploy CockroachDB to production, you need the following infrastructure.
+
+#### Nodes
+
+3+ machines that can communicate with one another.
+
+- Use at least three nodes to ensure that a majority of replicas (2/3) remains available if a node fails. Configurations with odd numbers of replicas are more robust than those with even numbers. Clusters of three and four nodes can each tolerate one node failure and still reach a majority (2/3 and 3/4 respectively), so the fourth replica doesn't add any extra fault-tolerance. To survive two simultaneous failures, you must have five replicas.
+
+- Run each node on a separate machine. Since CockroachDB replicates across nodes, running more than one node per machine increases the risk of data loss if a machine fails. Likewise, if a machine has multiple disks or SSDs, run one node with multiple `--store` flags and not one node per disk. For more details about stores, see [Start a Node](start-a-node.html).
+
+- Ideally, these machines should be located in you would want these machines located in separate locations to maximize survivability.
+
+#### Load Balancer
+
+A highly available load balancer (such as those provided by Google Cloud or through an orchestration tool like Kubernetes). You'll use this to connect to multiple nodes in your cluster; you don't want to rely on a single node, because if it goes down, your application will go down despite having other available nodes.
+
+#### Service Discovery
+
+Service discovery (such as Zookeeper or through an orchestration tool like Kubernetes) so new nodes can be found.
+
+#### Monitoring
+
+It's crucial to know when machines fail and be alerted. While CockroachDB provides an Admin UI with this information, you will also want to integrate your cluster with a tool like Prometheus to alert your team of outages.
+
+## Platform
+
+To deploy CockroachDB, it's important to understand the platform on which you plan to deploy it. 
+
+### Networking
+
+While CockroachDB can easily be deployed on Google Cloud, AWS, Microsoft Azure, or Digital Ocean, each of them has idiosyncrasies, particularly around how they handle networking. Important elements of your network to understand include:
+
+- Hostnames
+- Firewalls
+
+#### Hostnames
+
+CockroachDB requires nodes to communicate with one another, as well as accept client requests from your application. This is often handled by hostnames, but it's important to understand what those are on your platform, or if they're accessible network names.
+
+Depending on how you plan to deploy Cockroach, you need to ensure that the following hostnames resolve:
+
+- **Client communication** to let your application or load balancer communicate with your cluster. This defaults to the node's `$HOSTNAME`, but can be controlled with `cockroach start`'s `--host` flag.
```

Comment:
you should mention why we may want to use one vs another. eg: hostnames valid within the cluster due to a private subnet, vs hostnames that can be used from outside the subnet.